### PR TITLE
{cae}[foss/2022a] OpenFOAM vv2206, METIS v5.1.0, SCOTCH v7.0.1 w/ int64

### PR DESCRIPTION
**Pull request by WilleBell**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ('flex', '2.6.4'),
+]
+
+dependencies = [
+    ('libreadline', '8.1.2'),
+    ('ncurses', '6.3'),
+    ('METIS', '5.1.0', '-64bitint'),
+    ('SCOTCH', '7.0.1', '-64bitint'),
+    ('KaHIP', '3.14'),
+    ('CGAL', '4.14.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb:42]: @WilleBell Let's add a comment here that explains why Paraview is not included as a dependency, like it typically is in OpenFOAM easyconfigs.

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@WilleBell There's two options to make the test suite happy:

1) We add an exception for METIS/SCOTCH/OpenFOAM with a versionsuffix like `-64bit-int`;
2) We make the `OpenFOAM` easyconfig a bundle of METIS/SCOTCH/OpenFOAM as components.
    That sounds scary, but it's actually relatively trivial. That way we don't need separate easyconfigs for `METIS` & `SCOTCH`.

Let me know if you think option 2 makes sense, and if you need any help with it.
You may need to fiddle with `$LIBRARY_PATH` and `%(installdir)s` in the OpenFOAM component to make sure the libraries METIS/SCOTCH components are picked up when building OpenFOAM...

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name = 'OpenFOAM'
+version = 'v2206'
+versionsuffix = '-64bitint'
```
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb:3]: `-64bit-int` may be confusing if combined a versionsuffix with 64-bit floats.

maybe just `-int64`, which combined with 64-bit floats may become `64bit-int64`, as was done for `__archive__/a/ACML/ACML-5.2.0-gfortran-64bit-int64.eb`?

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb (hunk 1 above)
Comment (WilleBell) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb]: Paraview was added as dependency

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb (hunk 2 above)
[the remark this answers is not in the document]
Comment (WilleBell) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2206-foss-2022a-64bitint.eb]: Being the versionsuffix for 64-bit floats `-64bit` it seems less confusing to go for `-int64`. Let me know if you see a problem with this choice @boegel

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in total)
node3171.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/69afc79a7582f079b76031d917472180 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3138.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/113bbef00952957842ac57d73729053f for a full test report.

**Comment by boegel** [comment]
Going in, thanks @WilleBell!

**Pull request merged**